### PR TITLE
[feat] Support custom tetkon catalog

### PR DESCRIPTION
**Pull request by S-hayeon**
# Changes

Users can employ tekton catalogs either from tekton hub or users' git repo.
Valid argument is following
- public  repo: `public@{URL}` 
- private repo: `private@{URL}`
- Name w/ Version:  `{NAME}@{VERSION}`

When trying to employ catalog from private repository, personal-access-token from git is required and should be stored as secret under the name of  `private-access-token-{NAME}` and `access-token` for key value. Lastly, secret should be listed on workspaces.
The example is listed on [integration_config.md](https://github.com/tmax-cloud/cicd-operator/blame/6ee92232783fabfb54b5b73099f5d9e1ecaa3b38/docs/integration_config.md#L274)
related issue #336 

# Submitter Checklist

- [x] Docs included if needed
- [ ] Tests included if needed
- [x] Follows the [good commit messages standard](https://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by changjjjjjjj** [comment]
private repository 로직 추가후에 검토하겠습니다!

**Comment by S-hayeon** [comment]
the test fails because the personal access token is explicitly specified, causing to delete token immediately ([link](https://github.community/t/personal-access-token-deleting-itself/13955/2))

File: pkg/pipelinemanager/pipelinemanager.go [hunk 1]
```diff
@@ -78,11 +78,15 @@ func NewPipelineManager(c client.Client, s *runtime.Scheme) PipelineManager {
...
 func (p *pipelineManager) Generate(job *cicdv1.IntegrationJob) (*tektonv1beta1.PipelineRun, error) {
 	log.Info("Generating a pipeline run")
-
+	// token for private repo
+	var token string
 	// Workspace defs
 	var workspaceDefs []tektonv1beta1.PipelineWorkspaceDeclaration
 	for _, w := range job.Spec.Workspaces {
 		workspaceDefs = append(workspaceDefs, tektonv1beta1.PipelineWorkspaceDeclaration{Name: w.Name})
+		if w.Secret.SecretName == "private-access-token" {
```
Comment (changjjjjjjj) [inline pkg/pipelinemanager/pipelinemanager.go:87]: secret name이 무조건 private-access-token으로 고정시키면 private repo가 여러개일 때는 문제가 있을 것 같습니다.

Comment (S-hayeon) [reply pkg/pipelinemanager/pipelinemanager.go]: personal-access-token임을 특정할 수 있어야 하기 때문에 private-access-token-$(NAME) 형식으로 수정했습니다.

**Review by changjjjjjjj — APPROVED** [review]

**Pull request merged**